blog/2020-01-angular-scully.md

```diff
@@ -0,0 +1,332 @@
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent implements OnInit {
+  posts$: Observable<ScullyRoute[]>;
+
+  constructor(private srs: ScullyRoutesService) {}
+
+  ngOnInit() {
+    this.posts$ = this.srs.available$;
+  }
+}
+```
+
+To display the results we can simply use `ngFor` with the `async` pipe and list the results.
+A `ScullyRoute` will give us the routing path inside the `route` key and all other markdown meta data inside their appropriate key names.
+So we can extend for example our markdown meta data block with more keys (e.g. `thumbnail: assets/thumb.jpg`) and we can access them via those (`blog.thumbnail`) in our case.
+Our `app.component.html` can look like this:
+
+```html
+<ul>
+  <li *ngFor="let post of posts$ | async">
+    <a [routerLink]="post.route">{{ post.title }}</a>
+  </li>
+</ul>
+
+<hr />
+
+<router-outlet></router-outlet>
+```
+
+This will give us a fully routed blog page:
+
+![a simple blog created with scully](/assets/images/blog/scully-blog.gif)
+
+The `ScullyRoutesService` will return always all matching routes.
+In fact if we are creating more routes in our app, it will detect them and the result of the `available$` observable will contain their data.
+To list only blog posts from the `blog` route and directory we can just filter the result:
+
+```ts
+/* ... */
+import { map } from 'rxjs/operators';
+/* ... */
+export class AppComponent implements OnInit {
+  /* ... */
+  ngOnInit() {
+    this.posts$ = this.srs.available$.pipe(
+      map(routeList => {
+        return routeList.filter((route: ScullyRoute) =>
+          route.route.startsWith(`/blog/`),
+        );
+      })
+    );
+  }
+}
+```
+
+Wow! That was easy wasn’t it? Now you just need to add a bit of styling and content and your blog is ready for getting visited.
+
+## Fetch dynamic information from an API
+
+As you may have realized: _Scully_ needs a data source to fetch all dynamic routes in an app.
+In case of our blog example _Scully_ uses the `:slug` router param as a placeholder.
+Scully will fill this placeholder with appropriate content to visit and pre-render the site.
+The content for the placeholder comes in our blog example from the files in the `/blog` directory.
+This has been configured from the schematics we ran before in the file `scully.config.js`:
+
+```js
+exports.config = {
+  projectRoot: "./src/app",
+  routes: {
+    '/blog/:slug': {
+      type: 'contentFolder',
+      slug: {
+        folder: "./blog"
+      }
+    },
+  }
+};
+```
+
+Let’s imagine our app has another route called `/books/:isbn`.
+To visit this route and pre-render it, we need a way to fill the `isbn` parameter.
+Luckily _Scully_ helps us with this too.
+We can configure a — so called [_Route Plugin_](https://github.com/scullyio/scully/blob/master/docs/plugins.md) — that will call an API, fetch the data from it and pluck the `isbn` from the array of results to fill it in the router parameter.
```

Comment:
`We a so-called Router Plugin`. 

They are called `Router Plugins`. Check the docs, we updated the official name of them to Router Plugins. Check it out.